File: we_10_2_a_loose_coupling_position_update.m
% Integrated Navigation Systems — Chapter 10: GNSS/INS Integration Architectures
% Worked Example 10.2 — A loose-coupling position update
% Companion code (educational reference implementation; see the book for context).

Pm=100; R=25; z=12.0; H=1;
K=Pm*H/(H*Pm*H+R); dx=K*z; Pp=(1-K*H)*Pm;
fprintf('K=%.2f dx=%.1f P+=%.0f\n', K, dx, Pp);   % 0.80 9.6 20

File: we_10_2_a_loose_coupling_position_update.py
# Integrated Navigation Systems — Chapter 10: GNSS/INS Integration Architectures
# Worked Example 10.2 — A loose-coupling position update
# Companion code (educational reference implementation; see the book for context).

Pm,R,z,H=100,25,12.0,1
K=Pm*H/(H*Pm*H+R); print(f'K={K:.2f} dx={K*z:.1f} P+={(1-K*H)*Pm:.0f}')  # 0.80 9.6 20
